src/indexer/allo/v2/handleEvent.ts

```diff
@@ -1,10 +1,36 @@
 import { EventHandlerArgs } from "chainsauce";
+import { ethers } from "ethers";
 import type { Indexer } from "../../indexer.js";
-import { ProjectTable } from "../../../database/schema.js";
+import { ProjectTable, NewRound } from "../../../database/schema.js";
 import { Changeset } from "../../../database/index.js";
 import { parseAddress } from "../../../address.js";
 import roleGranted from "./roleGranted.js";
 import roleRevoked from "./roleRevoked.js";
+import { fetchPoolMetadata } from "./parsePoolMetadata.js";
+
+function padBytes32Hex(s: string): string {
+  if (s.length > 64) {
+    return s;
+  }
+
+  const padding = 64 - s.length;
+  let hex = s;
+  hex = "0".repeat(padding) + hex;
+  return "0x" + hex;
+}
+
+function generateRoundRoles(poolId: string) {
+  // POOL_MANAGER_ROLE = bytes32(poolId);
+  const managerRole = padBytes32Hex(poolId);
+
+  // POOL_ADMIN_ROLE = keccak256(abi.encodePacked(poolId, "admin"));
+  const adminRawRole = ethers.utils.solidityPack(
+    ["uint256", "string"],
+    [poolId, "admin"]
+  );
+  const adminRole = ethers.utils.solidityKeccak256(["bytes"], [adminRawRole]);
```

Comment:
https://viem.sh/docs/utilities/keccak256#keccak256

Reply:
cool, I updated it thank you!